test/depots/depotDonneesAutorisations.spec.js

```diff
@@ -258,4 +260,74 @@ describe('Le dépôt de données des autorisations', () => {
+      const adaptateurPersistance = AdaptateurPersistanceMemoire.nouvelAdaptateur({
+        utilisateurs: [
+          { id: '999', email: 'jean.dupont@mail.fr' },
+          { id: '000', email: 'annie.dubois@mail.fr' },
+        ],
+        homologations: [{ id: '123', descriptionService: { nomService: 'Un service' } }],
+        autorisations: [
+          { id: '456', idUtilisateur: '999', idHomologation: '123', type: 'createur' },
+        ],
+      });
+      const depot = DepotDonneesAutorisations.creeDepot({ adaptateurPersistance });
+
+      depot.supprimeContributeur('000', '123')
+        .then(() => done('La demande de suppression aurait dû lever une erreur'))
+        .catch((e) => {
+          expect(e).to.be.an(ErreurAutorisationInexistante);
+          expect(e.message).to.equal("L'utilisateur \"000\" n'est pas contributeur de l'homologation \"123\"");
+          done();
+        })
+        .catch(done);
+    });
+
+    it("vérifie qu'il s'agit bien d'un contributeur et non du créateur de l'homologation", (done) => {
+      const adaptateurPersistance = AdaptateurPersistanceMemoire.nouvelAdaptateur({
+        utilisateurs: [
+          { id: '999', email: 'jean.dupont@mail.fr' },
+        ],
+        homologations: [{ id: '123', descriptionService: { nomService: 'Un service' } }],
+        autorisations: [
+          { id: '456', idUtilisateur: '999', idHomologation: '123', type: 'createur' },
+        ],
+      });
+      const depot = DepotDonneesAutorisations.creeDepot({ adaptateurPersistance });
+
+      depot.supprimeContributeur('999', '123')
+        .then(() => done('La demande de suppression aurait dû lever une erreur'))
+        .catch((e) => {
+          expect(e).to.be.an(ErreurTentativeSuppressionCreateur);
+          expect(e.message).to.equal("Suppression impossible : l'utilisateur \"999\" est le propriétaire de l'homologation \"123\"");
+          done();
+        })
+        .catch(done);
+    });
+
+    it('supprime le contributeur', (done) => {
+      const adaptateurPersistance = AdaptateurPersistanceMemoire.nouvelAdaptateur({
+        utilisateurs: [
+          { id: '999', email: 'jean.dupont@mail.fr' },
+          { id: '000', email: 'contributeur@mail.fr' },
+        ],
+        homologations: [{ id: '123', descriptionService: { nomService: 'Un service' } }],
+        autorisations: [
+          { id: '456', idUtilisateur: '999', idHomologation: '123', type: 'createur' },
+          { id: '789', idUtilisateur: '000', idHomologation: '123', type: 'contributeur' },
+        ],
+      });
+      const depot = DepotDonneesAutorisations.creeDepot({ adaptateurPersistance });
+
+      depot.autorisationPour('000', '123')
+        .then((a) => expect(a).to.be.an(AutorisationContributeur))
+        .then(() => depot.supprimeContributeur('000', '123'))
+        .then(() => depot.autorisationPour('000', '123'))
+        .then((a) => expect(a).to.be(undefined))
```

Comment:
je préfère l'écriture _expect(a).to.not.be.ok()_ car l'égalité stricte undefined est trop lourde à mon gout, sachant que dans ce cas on assigne undefined à une variable et non pas une variable est undefined

Reply:
Vu en binômage :
- Je préfère qu'on teste ce qui est, plutôt que ce qui n'est pas
- Je ne suis pas sûr de voir l'intérêt de passer une valeur `null` plutôt que `undefined`, sachant que c'est la convention de knex. En en même temps… 🤷